plugins/modules/system/interfaces_file.py

```diff
@@ -148,57 +148,44 @@
 from ansible.module_utils.common.text.converters import to_bytes
 
 
-def lineDict(line):
+def line_dict(line):
     return {'line': line, 'line_type': 'unknown'}
 
 
-def optionDict(line, iface, option, value, address_family):
+def make_option_dict(line, iface, option, value, address_family):
     return {'line': line, 'iface': iface, 'option': option, 'value': value, 'line_type': 'option', 'address_family': address_family}
 
 
-def getValueFromLine(s):
-    spaceRe = re.compile(r'\s+')
-    for m in spaceRe.finditer(s):
-        pass
-    valueEnd = m.start()
-    option = s.split()[0]
-    optionStart = s.find(option)
-    optionLen = len(option)
-    valueStart = re.search(r'\s', s[optionLen + optionStart:]).end() + optionLen + optionStart
-    return s[valueStart:valueEnd]
+def get_option_value(line):
+    patt = re.compile(r'^\s+(?P<option>\S+)\s+(?P<value>\S?.*\S)\s*$')
+    match = patt.match(line)
+    if not match:
+        return None, None
+    return match.group("option"), match.group("value")
```

Comment:
The new function behaves quite differently from the old one (and I don't mean that it doesn't include the option name):
```
import re

def getValueFromLine(s):
    spaceRe = re.compile(r'\s+')
    for m in spaceRe.finditer(s):
        pass
    valueEnd = m.start()
    option = s.split()[0]
    optionStart = s.find(option)
    optionLen = len(option)
    valueStart = re.search(r'\s', s[optionLen + optionStart:]).end() + optionLen + optionStart
    return s[valueStart:valueEnd]

def get_option_value(line):
    patt = re.compile(r'^\s+(?P<option>\S+)\s+(?P<value>\S?.*\S)\s*$')
    match = patt.match(line)
    if not match:
        return None, None
    return match.group("option"), match.group("value")

print([(getValueFromLine(s), get_option_value(s)) for s in ['a b c', ' a b c', ' a b c ', ' a  b c ', 'a b c ', '  aaa  bbb  ccc  ddd']])
```
produces
```
[
    ('b', (None, None)),
    ('b', ('a', 'b c')),
    ('b c', ('a', 'b c')),
    (' b c', ('a', 'b c')),
    ('b c', (None, None)),
    (' bbb  ccc', ('aaa', 'bbb  ccc  ddd'))
]
```
(with manual pretty-printing). The first tuple element should be equal to the second element of the second tuple in every line, if these functions would behave the same.

Reply:
Yes, that's on purpose - the reason why it works is, perhaps, clearer if you look at where it is used:
https://github.com/russoz/community.general/blob/interfaces_file-improvement/plugins/modules/system/interfaces_file.py#L213-L215